examples/cmd/main.go

```diff
@@ -30,11 +32,52 @@ func main() {
 	app.RunMain()
 }
 
+type DownloadersPool struct {
+	Size int
+	pool chan *github.Downloader
+}
+
+func NewDownloadersPool(clients []*http.Client, db *sql.DB) (*DownloadersPool, error) {
+	ch := make(chan *github.Downloader, len(clients))
+
+	var builder func(*http.Client) (*github.Downloader, error)
+	if db == nil {
+		builder = github.NewStdoutDownloader
+	} else {
+		builder = func(c *http.Client) (*github.Downloader, error) {
+			return github.NewDownloader(c, db)
+		}
+	}
+
+	for _, c := range clients {
+		d, err := builder(c)
+		if err != nil {
+			return nil, err
+		}
+
+		ch <- d
+	}
+
+	return &DownloadersPool{
+		Size: len(clients),
+		pool: ch,
+	}, nil
+}
+
+func (dp *DownloadersPool) WithDownloader(f func(d *github.Downloader) error) error {
```

Comment:
Initially I implemented an alternative of this as follows:
```
func (dp *DownloadersPool) Acquire() (*github.Downloader, func()) {
	downloader := <-dp.pool
	release := func() {
		dp.pool <- downloader
	}
	return downloader, release
}
```

that should be used as:
```
d, release := dp.Acquire()
...do something with `d`...
release()
```

I actually liked this signature as it is similar to `context.Context`. But the problem is that even after calling `release`, the `d` object could still be used even if it's not supposed to. This could have been fixed by adding a state to the `github.Downloader`, but I preferred not to.